File: regiondeployer/CloudError.cs
using Microsoft.Azure.Management.Network.Fluent.Models;
using Microsoft.Azure.Management.AppService.Fluent.Models;
using System;
using Microsoft.Rest.Azure;
using System.Threading.Tasks;
using serversidearchitecture.common;
using MoonMachine.Core;

public class CloudErrorHelper
{
    public void PrintCloudError(Exception inputError, Action<string> logMessage)
    {
        var errorUtils = new ErrorUtilities();
        var log = errorUtils.CreateLog(inputError, string.Empty, false);
        var logOutput = string.Concat(log.Message, log.Stacktrace);
        logMessage(logOutput);
    }

    public void RaiseCloudError(Exception inputError)
    {
        var errorUtils = new ErrorUtilities();
        var log = errorUtils.CreateLog(inputError, string.Empty, false);
        var logOutput = string.Concat(log.Message, log.Stacktrace);
        throw new Exception(logOutput);
    }
}

File: regiondeployer/clouderror.fs
module regiondeployer

open Microsoft.Azure.Management.Network.Fluent.Models
open Microsoft.Azure.Management.AppService.Fluent.Models
open System
open Microsoft.Rest.Azure
open System.Threading.Tasks
open serversidearchitecture.common
open MoonMachine.Core

let public printclouderror(inputerror:Exception, logmessage:string -> unit) : unit =
    let errorutils : errorutilities = new errorutilities()
    let log = errorutils.createlog(inputerror, String.Empty, false)
    String.concat valueseparator [log.Message; log.Stacktrace]
    |> logmessage    

let public raiseclouderror(inputerror:Exception) : unit =
    let errorutils : errorutilities = new errorutilities()
    let log = errorutils.createlog(inputerror, String.Empty, false)
    String.concat valueseparator [log.Message; log.Stacktrace]
    |> failwith